src/main/kotlin/org/malachite/estella/services/TaskStageService.kt

```diff
@@ -108,19 +108,20 @@ class TaskStageService(
 
 
     fun addResult(resultToAdd: TaskService.ResultToAdd) {
-        val resultToSave = taskResultRepository.findAll().find { isResultNew(it, resultToAdd) }.let {
-            copyTaskResult(it, resultToAdd)
-                ?: createNewTaskResult(resultToAdd)
-        }
-        val savedResult = taskResultRepository.save(resultToSave)
-        val taskStage = savedResult.taskStage
-        val newTaskResults = taskStage!!.tasksResult.filter { it.task.id != resultToSave.task.id }.plus(savedResult)
-        taskStageRepository.save(taskStage.copy(tasksResult = newTaskResults.toSet()))
+        taskResultRepository.findAll().find { it.task.id == resultToAdd.task.id && it.taskStage!!.id == resultToAdd.taskStage.id }
+                .let {
+                    copyTaskResult(it, resultToAdd)
+                }?.let { result ->
+                    if (isFirstSolvedTask(resultToAdd) && shouldNotifyDev(resultToAdd))
+                        sendDevNotification(resultToAdd)
+
+                    taskResultRepository.save(result)
+                    val taskStage = result.taskStage!!
+                    val newTaskResults = taskStage.tasksResult.filter { it.task.id != result.task.id }.plus(result).toSet()
```

Comment:
Czy to oznacza że wywalamy stare TaskResult dla tego taska?

Reply:
tak

Reply:
@Davenury nie proponował tego żeby zapisywać wszystkie resulty?

Reply:
Z tego co kojarzę to gadałem z nim o tym i ustaliliśmy ze będziemy nadpisywać